internal/services/groups/groups_data_source.go

```diff
@@ -102,13 +119,15 @@ func groupsDataSourceRead(ctx context.Context, d *schema.ResourceData, meta inte
 			}
 
 			count := len(*result)
-			if count > 1 {
+
+			//Could we make this a switch case?
```

Comment:
If think this is fine as an if-else block for now. Maybe one more case would make it eligible IMO